sum array null pointer error

```
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
  File "/mnt/lustre/tpoterba/hail-inst/python/pyhail/dataset.py", line 77, in annotate_samples_expr
    return self.hc.run_command(self, pargs)
  File "/mnt/lustre/tpoterba/hail-inst/python/pyhail/context.py", line 45, in run_command
    cmd_args)
  File "/opt/spark/spark-1.5.2/python/lib/py4j-0.8.2.1-src.zip/py4j/java_gateway.py", line 538, in __call__
  File "/opt/spark/spark-1.5.2/python/pyspark/sql/utils.py", line 36, in deco
    return f(*a, **kw)
  File "/opt/spark/spark-1.5.2/python/lib/py4j-0.8.2.1-src.zip/py4j/protocol.py", line 300, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling o417.run.
: org.apache.spark.SparkDriverExecutionException: Execution error
    at org.apache.spark.scheduler.DAGScheduler.handleTaskCompletion(DAGScheduler.scala:1024)
    at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1493)
    at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1458)
    at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1447)
    at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
    at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:567)
    at org.apache.spark.SparkContext.runJob(SparkContext.scala:1824)
    at org.apache.spark.SparkContext.runJob(SparkContext.scala:1944)
    at org.apache.spark.rdd.RDD$$anonfun$reduce$1.apply(RDD.scala:1007)
    at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:147)
    at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:108)
    at org.apache.spark.rdd.RDD.withScope(RDD.scala:310)
    at org.apache.spark.rdd.RDD.reduce(RDD.scala:989)
    at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1.apply(RDD.scala:1118)
    at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:147)
    at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:108)
    at org.apache.spark.rdd.RDD.withScope(RDD.scala:310)
    at org.apache.spark.rdd.RDD.treeAggregate(RDD.scala:1095)
    at org.broadinstitute.hail.methods.Aggregators$.buildSampleAggregations(Aggregators.scala:66)
    at org.broadinstitute.hail.driver.AnnotateSamplesExpr$.run(AnnotateSamplesExpr.scala:63)
    at org.broadinstitute.hail.driver.AnnotateSamplesExpr$.run(AnnotateSamplesExpr.scala:11)
    at org.broadinstitute.hail.driver.Command.runCommand(Command.scala:258)
    at org.broadinstitute.hail.driver.Command.run(Command.scala:263)
    at sun.reflect.GeneratedMethodAccessor37.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:231)
    at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:379)
    at py4j.Gateway.invoke(Gateway.java:259)
    at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:133)
    at py4j.commands.CallCommand.execute(CallCommand.java:79)
    at py4j.GatewayConnection.run(GatewayConnection.java:207)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
    at org.broadinstitute.hail.methods.SumArrayAggregator.combOp(Aggregators.scala:348)
    at org.broadinstitute.hail.methods.SumArrayAggregator.combOp(Aggregators.scala:317)
    at org.broadinstitute.hail.methods.Aggregators$$anonfun$6$$anonfun$apply$1$$anonfun$apply$mcVI$sp$2.apply$mcVI$sp(Aggregators.scala:85)
    at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:141)
    at org.broadinstitute.hail.methods.Aggregators$$anonfun$6$$anonfun$apply$1.apply$mcVI$sp(Aggregators.scala:83)
    at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:141)
    at org.broadinstitute.hail.methods.Aggregators$$anonfun$6.apply(Aggregators.scala:83)
    at org.broadinstitute.hail.methods.Aggregators$$anonfun$6.apply(Aggregators.scala:82)
    at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$15.apply(RDD.scala:1002)
    at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$15.apply(RDD.scala:999)
    at org.apache.spark.scheduler.JobWaiter.taskSucceeded(JobWaiter.scala:56)
    at org.apache.spark.scheduler.DAGScheduler.handleTaskCompletion(DAGScheduler.scala:1020)
    at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1493)
    at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1458)
    at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1447)
    at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
```

--- a/src/main/scala/org/broadinstitute/hail/methods/Aggregators.scala
+++ b/src/main/scala/org/broadinstitute/hail/methods/Aggregators.scala
@@ -258,8 +258,7 @@ class CounterAggregator extends TypedAggregator[IndexedSeq[Annotation]] {
   }.toArray[Annotation]: IndexedSeq[Annotation]
 
   def seqOp(x: Any) {
-    if (x != null)
-      m.updateValue(x, 0L, _ + 1)
+    m.updateValue(x, 0L, _ + 1)
   }
 
   def combOp(agg2: this.type) {
@@ -297,8 +296,7 @@ class CollectAggregator extends TypedAggregator[ArrayBuffer[Any]] {
   def result = _state
 
   def seqOp(x: Any) {
-    if (x != null)
-      _state += x
+    _state += x
   }
 
   def combOp(agg2: this.type) = _state ++= agg2._state
@@ -393,12 +391,17 @@ class SumArrayAggregator[T](implicit ev: scala.math.Numeric[T], ct: ClassTag[T])
 
   def combOp(agg2: this.type) = {
     val agg2state = agg2._state
-    if (_state.length != agg2state.length)
-      fatal(
-        s"""cannot aggregate arrays of unequal length with `sum'
-            |  Found conflicting arrays of size (${ _state.length }) and (${ agg2state.length })""".stripMargin)
-    for (i <- _state.indices)
-      _state(i) += agg2state(i)
+    if (_state == null)
+      _state = agg2._state
+    else if (agg2._state != null) {
+      if (_state.length != agg2state.length)
+        fatal(
+          s"""cannot aggregate arrays of unequal length with `sum'
+              |  Found conflicting arrays of size (${ _state.length }) and (${ agg2state.length })""".
+            stripMargin)
+      for (i <- _state.indices)
+        _state(i) += agg2state(i)
+    }
   }
 
   def copy() = new SumArrayAggregator()

--- a/src/main/scala/org/broadinstitute/hail/methods/Aggregators.scala
+++ b/src/main/scala/org/broadinstitute/hail/methods/Aggregators.scala
@@ -215,7 +215,6 @@ class FractionAggregator(f: (Any) => Any)
     _denom += 1
     if (r.asInstanceOf[Boolean])
       _num += 1
-    }
   }
 
   def combOp(agg2: this.type) {
@@ -503,9 +502,9 @@ class TakeByAggregator[T](var f: (Any) => Any, var n: Int)(implicit var tord: Or
   // https://issues.scala-lang.org/browse/SI-7568
   // fixed in Scala 2.11.0-M7
   var _state = if (ord != null)
-      new mutable.PriorityQueue[(Any, Any)]()(ord)
-    else
-      null
+    new mutable.PriorityQueue[(Any, Any)]()(ord)
+  else
+    null
 
   def result = _state.toArray[(Any, Any)].map(_._1).reverse: IndexedSeq[Any]
 

--- a/src/main/scala/org/broadinstitute/hail/methods/Aggregators.scala
+++ b/src/main/scala/org/broadinstitute/hail/methods/Aggregators.scala
@@ -188,8 +188,7 @@ class CountAggregator() extends TypedAggregator[Long] {
   def result = _state
 
   def seqOp(x: Any) {
-    if (x != null)
-      _state += 1
+    _state += 1
   }
 
   def combOp(agg2: this.type) {
@@ -213,10 +212,9 @@ class FractionAggregator(f: (Any) => Any)
 
   def seqOp(x: Any) {
     val r = f(x)
-    if (r != null) {
-      _denom += 1
-      if (r.asInstanceOf[Boolean])
-        _num += 1
+    _denom += 1
+    if (r.asInstanceOf[Boolean])
+      _num += 1
     }
   }
 

